sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1415,6 +1415,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val USE_DEPRECATED_KAFKA_OFFSET_FETCHING =
+    buildConf("spark.sql.streaming.kafka.useDeprecatedOffsetFetching")
+      .internal()
+      .doc("When true, the deprecated Consumer based offset fetching used which could cause " +
```

Comment:
As we only describe the details in migration doc which is not something end users always read as a guide doc or manual doc, we'd be better to write explanation either here or sql+kafka integration doc.
(IMO it worths to add the comparison in the integration doc, and let here refer the doc.)

Reply:
Fixed. Updating the description because I've double checked the generated HTML files too.